### PR TITLE
What's New documentation for v2.3.0

### PR DESCRIPTION
**Pull request by denyeart**
#### Type of change

- Documentation update

#### Description

Add What's New documentation for v2.3.0.
Also mention the new features in the v2.3.0 release notes.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

### CODE REVIEW

**Review by yacovm — DISMISSED** [review]

File: docs/source/whatsnew.rst [hunk 1]
```diff
@@ -1,21 +1,74 @@
...
+* **Scalability** When there is a large number of ordering nodes and channels defined on the system channel,
+  it can take a long time for ordering nodes to reach consensus on the membership of all the channels.
+  Now, an ordering service can scale horizontally in a decentralized fashion by independently joining ordering nodes to specific channels.
+* **Operational benefits**
+   * Simple process to join an ordering node to a channel.
+   * Ability to list the channels that the ordering node is a consenter on.
+   * Simple process to remove a channel from an ordering node, which automatically cleans up the blocks associated with that channel.
+   * Peer organizations do not need to coordinate with an admin of the system channel to create or update its MSP.
+
+For more information, see the :doc:`create_channel/create_channel_participation` topic.
```
Comment (denyeart) [inline docs/source/whatsnew.rst:30]: @pamandrejko @joealewine Looks like the link didn't work, how to link to a topic in a subdirectory?

Comment (pamandrejko) [reply docs/source/whatsnew.rst:30]: I saw that too. I was thinking it did not work because that tutorial was not yet merged?

Comment (denyeart) [reply docs/source/whatsnew.rst:30]: Oh right, I've rebased and force pushed again...

File: docs/source/whatsnew.rst [hunk 2]
```diff
@@ -1,21 +1,74 @@
...
+To simplify the channel creation process and enhance the privacy and scalability of channels,
+it is now possible to create application channels without first creating a “system channel” managed by the ordering service.
+This process allows ordering nodes to join (or leave) any number of channels as needed, similar to how peers can participate in multiple channels.
+
+Benefits of the new process:
+
+* **Increased privacy** Because all ordering nodes used to be joined to the system channel,
+  every ordering node in a network knew about the existence of every channel on that ordering service.
+  Now, an ordering node only knows about the channels it is joined to.
+* **Scalability** When there is a large number of ordering nodes and channels defined on the system channel,
```
Comment (joealewine) [inline docs/source/whatsnew.rst]: When rendered, the bold word followed by an unbolded word looks weird. I'd add a colon here and throughout this doc (other than with "Operational benefits" --- the sublist makes it unnecessary).

**Review by wlahti — COMMENTED** [review]
Looks good to me.

File: docs/source/whatsnew.rst (hunk 2 above)
Comment (denyeart) [reply docs/source/whatsnew.rst]: Done

File: docs/source/whatsnew.rst [hunk 3]
```diff
@@ -1,21 +1,74 @@
...
+It is now possible to take a snapshot of a peer's channel information, including its state database,
+and join new peers (in the same organization or different organizations) to the channel based on the snapshot.
+
+Using ledger snapshots has the following advantages:
+
+* **Peers don't need to process all blocks since genesis block:** Peers can join a channel without processing all
+  previous blocks since the genesis block, greatly reducing the time it takes to join a peer to an existing channel.
+* **Peers can join channels using latest channel configuration:** Because snapshots include the latest channel configuration,
+  peers can now join a channel using the latest channel configuration.
+  This is especially important if critical channel configuration such as orderer endpoints or MSP CA certificates have been updated since the genesis block.
```
Comment (jyellick) [inline docs/source/whatsnew.rst]: A nit, but I'd say it's more critical if the TLS CA certificates have changed.  In general, different signing certs aren't troublesome.

**Review by jyellick — COMMENTED** [review]
Looks good to me, one very minor nit.

File: docs/source/whatsnew.rst (hunk 3 above)
Comment (denyeart) [reply docs/source/whatsnew.rst]: ah right, done.

**Comment by denyeart** [comment]
I'll rebase one more time before we merge, to pick up the latest release note changes.

**Review by manish-sethi — APPROVED** [review]

**Pull request merged**